packages/medusa-plugin-meilisearch/src/loaders/index.js

```diff
@@ -0,0 +1,66 @@
+import { transformProduct } from "../utils/transform-product"
```

Comment:
I think all of the logic in here should be moved to the core so that on load the products will be indexed no matter what. The responsibility of the indexing should then be taken care of by the installed searchService. Flow should be something like this:

1. in loader use `productService.list` to fetch batches of products
2. then call `searchService.addDocuments(index, productBatch)`
3. transformations etc will be taken care of by the search service
4. search service calls provider specific endpoints to index batch

It might be that it becomes a bit tricky with addDocuments having to do conditional transformations based on the index, it could make sense to extend the Search API in this case so that search plugins would have to implement a transformProducts method. This would allow us to add a step:

1b. call `searchService.transformProducts(productPatch)`

The return value of the transformProducts method could then be used directly when calling `addDocuments(index, transformedBatch)`.

Doing it this way would mean that the search plugin wouldn't have to be aware of the `INDEX_NS` variable which seems to be a path to mistakes.

We should maybe remove settings from the core Search API for now and allow plugins to control these themselves until we can find a more common approach?

Reply:
@srindom thanks for the great feedback!

> I think all of the logic in here ...

Does that include the logic for [updating index settings](https://github.com/medusajs/medusa/blob/560cc0de556eb966885979ea86ff1246f6f8251a/packages/medusa-plugin-meilisearch/src/loaders/index.js#L54-L60)? Just making sure because you later mentioned that the above flow would make the search plugin oblivious to `INDEX_NS`, which is not the case because we still need it to update the index settings. 

However, on a second (actually third or fourth :smile: ) thought, I totally agree with the fact that it is a path to mistakes.. So I suggest we completely remove the `INDEX_NS` for now. 

> We should maybe remove settings from the core Search API for now and allow plugins to control these themselves until we can find a more common approach?

Sorry, i'm not sure I got what settings you are referring to here :( 

> It might be that it becomes a bit tricky with addDocuments having to do conditional transformations based on the index, it could make sense to extend the Search API in this case so that search plugins would have to implement a transformProducts method. This would allow us to add a step:
> 
> 1b. call searchService.transformProducts(productPatch)

I'm trying to give this more thought. My concern is that we'll start extending the search service (SearchService interface + DefaultSearchService) in case we want to add more indices (and those need transformations). I'd personally rather let the plugins deal with this, or at the very least, use dedicated services that would act as mappers/transformers for example:
```javascript
// medusa-plugin-meilisearch/src/services/product-transformer.js
class ProductTransformer extends BaseService {
  transform(products) {
    ...
  }
}
```
However, If you think that adding more extending the `SearchService` regularly is a far-fetched scenario, then I think it's all good.